tensorflow/compiler/jit/xla_device_ops.h

```diff
@@ -72,6 +72,11 @@ class XlaAssignVariableOp : public OpKernel {
 #define REGISTER_XLA_RUN_KERNEL(DEVICE, KERNEL, TYPES) \
   REGISTER_KERNEL_BUILDER(Name("_XlaRun").Device(DEVICE), KERNEL);
 
+#define REGISTER_XLA_MERGE_KERNEL(DEVICE, KERNEL, TYPES)  \
```

Comment:
When do you need this?  Since XLA_* device clusters are always compiled, we should not need to generate merge ops for them.

Reply:
I'm not sure.
But how is op different from _XlaRun and _XlaCompile in this context?

Reply:
For ops placed on XLA_* devices we still create clusters as usual.  The semantic for XLA_* devices is that:

  1. Ops placed on these devices must be compiled
  2. Ops placed on these devices will be opportunistically clustered

Since ops placed on these devices **must** be compiled we do not need (rather cannot have) the lazy compilation scheme so we should be generating just the `_XlaCompile` and `_XlaRun` nodes placed on XLA_* devices.  Now that I think of it, it might be more logical to place these `_XlaCompile` and `_XlaRun` nodes on `GPU` and not `XLA_GPU`, but that's a separate change.

I also want to stress that the `XLA_*` devices are only there for unit tests, we usually push back on users trying to use them in regular code.